src/fieldAnalyser/typeUtils.ts

```diff
@@ -25,7 +25,22 @@ export function getMostEnglobingType(
   return null;
 }
 
+export function isMultiValueFacet(obj: unknown): boolean {
+  // Handling the most obvious use case. A multi value field could also be a string with delimiter, but we do not want to assume anything...
+  // With an array, we know for sure the metadata has multiple values
+  return Array.isArray(obj);
+}
+
+export function isHierarchicalFacet(obj: unknown): boolean {
+  // TODO: check if the array values follow a isHierarchy
+  const isHierarchical = false;
+  return isMultiValueFacet(obj) && isHierarchical;
+}
+
 export function getGuessedTypeFromValue(obj: unknown): FieldTypes {
+  if (Array.isArray(obj) && obj.length > 0) {
+    return getGuessedTypeFromValue(obj[0]); // FIXME: Not sure here. What should we use to define the type?
+  }
```

Comment:
What happens if we have something as ugly as `"field_name": [10.99, "string"]`?
Should it be considered as a type inconsistency (meaning an error will be thrown)?